src/nunavut/lang/py/support/nunavut_support.j2

```diff
@@ -0,0 +1,1802 @@
+        self.add_unaligned_bytes(self._float_to_bytes("d", x))
+
+    def add_unaligned_bit(self, x: bool) -> None:
+        self._buf[self._byte_offset] |= bool(x) << (self._bit_offset % 8)
+        self._bit_offset += 1
+
+    #
+    # Private methods.
+    #
+    @staticmethod
+    def _unsigned_to_bytes(value: int, bit_length: int) -> NDArray[Byte]:
+        assert bit_length >= 1
+        assert value >= 0, "This operation is undefined for negative integers"
+        value &= 2**bit_length - 1
+        num_bytes = (bit_length + 7) // 8
+        out: NDArray[Byte] = numpy.zeros(num_bytes, dtype=Byte)
+        for i in range(num_bytes):  # Oh, why is my life like this?
+            out[i] = value & 0xFF
+            value >>= 8
+        return out
+
+    @staticmethod
+    def _float_to_bytes(format_char: str, x: float) -> NDArray[Byte]:
+        f = "<" + format_char
+        try:
+            out = struct.pack(f, x)
+        except OverflowError:  # Oops, let's truncate (saturation must be implemented by the caller if needed)
+            out = struct.pack(f, numpy.inf if x > 0 else -numpy.inf)
+        # Note: this operation does not copy the underlying bytes
+        return numpy.frombuffer(out, dtype=Byte)
+
+    @staticmethod
+    def _ensure_not_negative(x: int) -> None:
+        if x < 0:
+            raise ValueError(f"The requested serialization method is not defined on negative integers ({x})")
+
+    @property
+    def _byte_offset(self) -> int:
+        return self._bit_offset // 8
+
+    def __str__(self) -> str:
+        s = " ".join(map(_byte_as_bit_string, self.buffer))
+        if self._bit_offset % 8 != 0:
+            s, tail = s.rsplit(maxsplit=1)
+            bits_to_cut_off = 8 - self._bit_offset % 8
+            tail = ("x" * bits_to_cut_off) + tail[bits_to_cut_off:]
+            return s + " " + tail
+        return s
+
+    def __repr__(self) -> str:
+        return f"{type(self).__name__}({self})"
+
+
+class _LittleEndianSerializer(Serializer):
+    # noinspection PyUnresolvedReferences
+    def add_aligned_array_of_standard_bit_length_primitives(self, x: NDArray[StdPrimitive]) -> None:
+        # This is close to direct memcpy() from the source memory into the destination memory, which is very fast.
+        # We assume that the local platform uses IEEE 754-compliant floating point representation; otherwise,
+        # the generated serialized representation may be incorrect. NumPy seems to only support IEEE-754 compliant
+        # platforms though so I don't expect any compatibility issues.
+        self.add_aligned_bytes(x.view(Byte))
+
+    def add_unaligned_array_of_standard_bit_length_primitives(self, x: NDArray[StdPrimitive]) -> None:
+        # This is much slower than the aligned version because we have to manually copy and shift each byte,
+        # but still better than manual elementwise serialization.
+        self.add_unaligned_bytes(x.view(Byte))
+
+
+class _BigEndianSerializer(Serializer):
```

Comment:
The reason this is provided here is because my plan was to take advantage of the highly dynamic, reflexive nature of Python and always generate one version of the support file, such that generated code could be used with any support file with the same major API version. I understand how this is undesirable for a high design assurance codebase but this Python implementation might benefit from a slightly different approach to make it closer to what one might call "idiomatic Python". Do you think it makes sense?

Reply:
okay